src/postgres/commands/createPostgresServer.ts

```diff
@@ -0,0 +1,63 @@
+import { ext } from '../../extensionVariables';
+import { SubscriptionTreeItem } from '../../tree/SubscriptionTreeItem';
+import { localize } from '../../utils/localize';
+import { nonNullProp } from '../../utils/nonNull';
+import { PostgresServerTreeItem } from '../tree/PostgresServerTreeItem';
+import { setFirewallRule } from './configurePostgresFirewall';
+import { IPostgresWizardContext } from './PostgresAccountWizard/IPostgresWizardContext';
+import { PostgresServerConfirmPWStep } from './PostgresAccountWizard/PostgresServerConfirmPWStep';
+import { PostgresServerCreateStep } from './PostgresAccountWizard/PostgresServerCreateStep';
+import { PostgresServerCredPWStep } from './PostgresAccountWizard/PostgresServerCredPWStep';
+import { PostgresServerCredUserStep } from './PostgresAccountWizard/PostgresServerCredUserStep';
+import { PostgresServerFirewallStep } from './PostgresAccountWizard/PostgresServerFirewallStep';
+import { PostgresServerNameStep } from './PostgresAccountWizard/PostgresServerNameStep';
+
+export async function createPostgresServer(context: IActionContext, node?: SubscriptionTreeItem): Promise<void> {
+    if (!node) {
+        node = await ext.tree.showTreeItemPicker<SubscriptionTreeItem>(SubscriptionTreeItem.contextValue, context);
+    }
+    const wizardContext: IPostgresWizardContext = Object.assign(context, node.root);
+    const promptSteps: AzureWizardPromptStep<ILocationWizardContext>[] = [
+        new PostgresServerNameStep(),
+        new ResourceGroupListStep(),
+        new PostgresServerCredUserStep(),
+        new PostgresServerCredPWStep(),
+        new PostgresServerConfirmPWStep(),
+        new PostgresServerFirewallStep()
+    ];
+
+    LocationListStep.addStep(wizardContext, promptSteps);
+    const wizard = new AzureWizard(wizardContext, {
+        promptSteps,
+        executeSteps: [
+            new PostgresServerCreateStep()
+        ],
+        title: localize('createPostgresServerPrompt', 'Create new PostgreSQL server')
+    });
+    await wizard.prompt();
+    const serverName: string = nonNullProp(wizardContext, 'newServerName');
+    await wizard.execute();
+
+    vscode.window.showInformationMessage(localize('createdServerMsg', 'Successfully created server "{0}".', wizardContext.newServerName));
+    const serverTree: PostgresServerTreeItem = new PostgresServerTreeItem(node, nonNullProp(wizardContext, 'server'));
+    let user: string = nonNullProp(wizardContext, 'adminUser');
+    const usernameSuffix: string = `@${serverName}`;
+    if (!user.includes(usernameSuffix)) {
+        user += usernameSuffix;
+    }
+    const password: string = nonNullProp(wizardContext, 'adminPassword');
+    void serverTree.setCredentials(user, password);
+    if (wizardContext.addFirewall) {
+        const ip: string = await publicIp.v4();
+        void setFirewallRule(serverTree, ip);
```

Comment:
Did you add `void` because you didn't want to await this? If so - why didn't you want to await this?

Also - we usually disable the lint rule instead of using `void` - it's much easier to tell at a glance what's going on:
```
// tslint:disable-next-line: no-floating-promises
```

Reply:
Actually I think we need await at serverTree.setCredentials and setFirewallRule, given they depend on wizard execution.